packages/containers/psammead-timestamp-container/src/index.stories.jsx

```diff
@@ -1,52 +1,137 @@
+  });
+  button('30 seconds ago', () => {
+    timestamp = moment().subtract(30, 'seconds');
+  });
+  button('59 seconds ago', () => {
+    timestamp = moment().subtract(59, 'seconds');
+  });
+  button('1 minute ago', () => {
+    timestamp = moment().subtract(1, 'minute');
+  });
+  button('24 minutes ago', () => {
+    timestamp = moment().subtract(24, 'minutes');
+  });
+  button('59 minutes ago', () => {
+    timestamp = moment().subtract(59, 'minutes');
+  });
+  button('63 minutes ago', () => {
+    timestamp = moment().subtract(63, 'minutes');
+  });
+  button('95 minutes ago', () => {
+    timestamp = moment().subtract(95, 'minutes');
+  });
+  button('2 hours ago', () => {
+    timestamp = moment().subtract(2, 'hours');
+  });
+  button('23 hours ago', () => {
+    timestamp = moment().subtract(23, 'hours');
+  });
+  button('1 day ago', () => {
+    timestamp = moment().subtract(1, 'day');
+  });
+  button('30 hours ago', () => {
+    timestamp = moment().subtract(30, 'hours');
+  });
+  button('20 days ago', () => {
+    timestamp = moment().subtract(20, 'days');
+  });
+  button('21 days ago', () => {
+    timestamp = moment().subtract(21, 'days');
+  });
+  button('8.9999... months ago', () => {
+    timestamp = moment()
+      .subtract(9, 'months')
+      .add(2, 'second');
+  });
+  button('9 months ago', () => {
+    timestamp = moment().subtract(9, 'months');
+  });
+
+  return number(name, timestamp.valueOf());
+};
+
 storiesOf('Containers|TimestampContainer', module)
   .addDecorator(withKnobs)
-  .add('default', () => (
-    <Timestamp
-      timestamp={number('Unix timestamp', 1530947227000)}
-      dateTimeFormat="YYYY-MM-DD"
-      format="D MMMM YYYY"
-      isRelative={boolean('isRelative', false)}
-      script={latin}
-      service="news"
-    />
-  ))
-  .add('with prefix', () => (
-    <Timestamp
-      timestamp={number('Unix timestamp', 1552666749637)}
-      dateTimeFormat="YYYY-MM-DD"
-      format="D MMMM YYYY, HH:mm z"
-      isRelative={boolean('isRelative', true)}
-      prefix={text('Prefix text', 'Updated')}
-      script={latin}
-      service="news"
-    />
-  ))
-  .add('with prefix and suffix', () => (
-    <Timestamp
-      timestamp={number('Unix timestamp', 1530947227000)}
-      dateTimeFormat="YYYY-MM-DD"
-      format="D MMMM YYYY, HH:mm z"
-      isRelative={boolean('isRelative', false)}
-      prefix={text('Prefix text', 'This')}
-      suffix={text('Suffix text', 'is date of last update')}
-      script={latin}
-      service="news"
-    />
-  ))
-  .add('with locale', () => (
-    <Timestamp
-      timestamp={number('Unix timestamp', 1530947227000)}
-      dateTimeFormat="YYYY-MM-DD"
-      format="D MMMM YYYY, HH:mm z"
-      isRelative={boolean('isRelative', false)}
-      script={latin}
-      locale={text('Locale', 'fa')}
-      service="persian"
-    />
-  ));
+  .add(
+    'default',
+    inputProvider([], ({ locale, script, service }) => (
```

Comment:
ideally this would use the dirDecorator, but that doesn't pass the locale through yet